sdk/python/kfp/compiler/compiler.py

```diff
@@ -946,4 +940,32 @@ def _create_and_write_workflow(
         params_list,
         pipeline_conf)
     self._write_workflow(workflow, package_path)
+    _validate_workflow(workflow)
```

Comment:
Since `_validate_workflow` is called after `_write_workflow`, does it mean even if the workflow cannot pass the validation the package is still dumped?

Reply:
I made this choice so that the defective compiled workflow can be sent in a bug report.
WDYT?

Reply:
I'm fine with it. It shall be captured by the backend anyway.